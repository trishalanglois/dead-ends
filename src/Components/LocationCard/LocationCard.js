import React from 'react';
import './LocationCard.scss';

export const LocationCard = ({ location, description }) => {

  return (
    <aside className="card">
      <h3 className="card-title">{location}</h3>
      <p className="card-description">{description}</p>
    </aside>
  )
}

export default LocationCard;
