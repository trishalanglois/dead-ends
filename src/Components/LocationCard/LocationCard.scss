@import '../../scss/variables';

.card {
  font-family: $main-font;
  width: 50em;
  margin: 2em;
  padding: 2em 2em 1em 2em;
  background-color: $light-grey;
}

.card-title {
  font-family: $main-font;
  color: $dark-color;
  font-size: 2em;
  margin-top: 0em;
}

.card-description {
  color: $dark-color;
  font-family: $main-font;
  font-size: 1.15em;
}

.card-title-container {
  display: flex;
  justify-content: space-between;
}

.card-button-container {
  display: flex;
  flex-direction: column;

}

.card-button {
  display: block;
  float: right;
  font-size: 1.25em;
  font-family: $main-font;
  margin: .25em;
  width: 100%;
  border-style: none;
  background-color: $medium-grey;
    &:hover {
      cursor: pointer;
      transform: scale(1.01);
      background-color: $dark-grey;
    }

}
