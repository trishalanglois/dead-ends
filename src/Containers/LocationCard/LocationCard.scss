@import '../../scss/variables';

* {
  border: 1px solid red;
}

.card {
  font-family: $main-font;
  width: 50em;
  margin: 0em 2em 2em 2em;
  padding: 2em 2em 1em 2em;
  background-color: $light-grey;
  border: 1.5px solid $dark-color;
}

.card-title {
  font-family: $main-font;
  color: $dark-color;
  font-size: 2em;
  margin-top: 0em;
}

.card-description {
  color: $dark-color;
  font-family: $main-font;
  font-size: 1.15em;
}

.card-title-container {
  display: flex;
  justify-content: space-between;
}

.button-wrapper {
  display: flex;
  flex-direction: column;
}

.card-button {
  float: right;
  font-size: 1.35em;
  font-family: $main-font;
  margin: .25em;
  padding: 0em 1em -1em 1em;
  width: 10em;
  height: 2em;
  border-style: none;
  background-color: $medium-grey;
    &:hover {
      cursor: pointer;
      transform: scale(1.01);
      background-color: $dark-grey;
    }
}

.remove-button {
  width: 11.7em !important;
  height: 2.5em !important;
  background-color: $wishlist-color !important;
  font-size: 1.15em !important;
}
