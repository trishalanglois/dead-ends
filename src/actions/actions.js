export const currentUser = user => ({
  type: 'SET_USER',
  name: user
})

export const setLocation = location => ({
  type: 'SET_LOCATION',
  location
})

export const setSpookyLocations = locations => ({
  type: 'SET_SPOOKY_LOCATIONS',
  locations
})

export const updateFavorites = location => ({
  type: 'UPDATE_FAVORITES',
  location
})

export const updateClickedCard = location => ({
  type: 'UPDATE_CLICKED_CARD',
  location
})
