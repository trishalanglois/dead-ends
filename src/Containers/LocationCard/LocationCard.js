import React from 'react';
import './LocationCard.scss';
import { updateFavorites } from '../../actions/actions';
import { connect } from 'react-redux';


export const LocationCard = ({ location, updateFavorites }) => {

  return (
    <aside className="card">
    <div className="card-title-container">
      <h3 className="card-title">{location.location}</h3>

      <div className="card-button-container">
        <button className="card-button" onClick={() => updateFavorites(location)}>Add to Favorites</button>
        <button className="card-button">View Map</button>
      </div>

    </div>
      <p className="card-description">{location.description}</p>
    </aside>
  )
}

export const mapDispatchToProps = dispatch => ({
  updateFavorites: newLocation => dispatch(updateFavorites(newLocation))
})

export default connect(null, mapDispatchToProps)(LocationCard);
