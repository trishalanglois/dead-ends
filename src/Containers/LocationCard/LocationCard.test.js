import React from 'react';
import { LocationCard, mapDispatchToProps, mapStateToProps } from './LocationCard';
import { updateFavorites, updateClickedCard, removeFavorite } from '../../actions/actions';
import { shallow } from 'enzyme';

describe('LocationCard', () => {
  let mockProps = {
      favorites:
        [{
          "city": "Denver",
          "country": "United States",
          "description": "BOO!",
          "location": "Ada Cemetery",
          "state": "Colorado",
          "state_abbrev": "CO",
          "longitude": -85.50489309999999,
          "latitude": 42.9621061,
          "city_longitude": -85.4954803,
          "city_latitude": 42.960727
        },
        {
          "city": "Muskegon",
          "country": "United States",
          "description": "SpooooOOOoooky!",
          "location": "Nunica Cemetery",
          "state": "Michigan",
          "state_abbrev": "MI",
          "longitude": -85.50489309999999,
          "latitude": 42.9621061,
          "city_longitude": -85.4954803,
          "city_latitude": 42.960727
        }],
      cardLocation: {
        "city": "Denver",
        "country": "United States",
        "description": "BOO!",
        "location": "Ada Cemetery",
        "state": "Colorado",
        "state_abbrev": "CO",
        "longitude": -85.50489309999999,
        "latitude": 42.9621061,
        "city_longitude": -85.4954803,
        "city_latitude": 42.960727
      }
    }

  it('should match the snapshot', () => {
    let wrapper = shallow(<LocationCard
      {...mockProps}
    />)

    expect(wrapper).toMatchSnapshot();
  })

  describe('mapStateToProps', () => {
    it('should return the favorite spooky locations from the global store', () => {
      const mockState = {
        favorites:
          [{
            "city": "Denver",
            "country": "United States",
            "description": "BOO!",
            "location": "Ada Cemetery",
            "state": "Colorado",
            "state_abbrev": "CO",
            "longitude": -85.50489309999999,
            "latitude": 42.9621061,
            "city_longitude": -85.4954803,
            "city_latitude": 42.960727
          },
          {
            "city": "Muskegon",
            "country": "United States",
            "description": "SpooooOOOoooky!",
            "location": "Nunica Cemetery",
            "state": "Michigan",
            "state_abbrev": "MI",
            "longitude": -85.50489309999999,
            "latitude": 42.9621061,
            "city_longitude": -85.4954803,
            "city_latitude": 42.960727
          }
        ]
      };
      const expected = {
        favorites:
          [{
            "city": "Denver",
            "country": "United States",
            "description": "BOO!",
            "location": "Ada Cemetery",
            "state": "Colorado",
            "state_abbrev": "CO",
            "longitude": -85.50489309999999,
            "latitude": 42.9621061,
            "city_longitude": -85.4954803,
            "city_latitude": 42.960727
          },
          {
            "city": "Muskegon",
            "country": "United States",
            "description": "SpooooOOOoooky!",
            "location": "Nunica Cemetery",
            "state": "Michigan",
            "state_abbrev": "MI",
            "longitude": -85.50489309999999,
            "latitude": 42.9621061,
            "city_longitude": -85.4954803,
            "city_latitude": 42.960727
          }
        ]
      };
      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {
    it('should call updateFavorites with a location when updateFavorites is called from props', () => {
      const mockDispatchFromProps = jest.fn();
      const actionToDispatch = updateFavorites(mockProps);
      const mappedProps = mapDispatchToProps(mockDispatchFromProps);
      mappedProps.updateFavorites(mockProps);

      expect(mockDispatchFromProps).toHaveBeenCalledWith(actionToDispatch);
    })

    it('should call updateClickedCard with a location when updateClickedCard is called from props', () => {
      const mockDispatchFromProps = jest.fn();
      const actionToDispatch = updateClickedCard(mockProps);
      const mappedProps = mapDispatchToProps(mockDispatchFromProps);
      mappedProps.updateClickedCard(mockProps);

      expect(mockDispatchFromProps).toHaveBeenCalledWith(actionToDispatch);
    })

    it('should call removeFavorite with a location when removeFavorite is called from props', () => {
      const mockDispatchFromProps = jest.fn();
      const actionToDispatch = removeFavorite(mockProps);
      const mappedProps = mapDispatchToProps(mockDispatchFromProps);
      mappedProps.removeFavorite(mockProps);

      expect(mockDispatchFromProps).toHaveBeenCalledWith(actionToDispatch);
    })
  })
})
