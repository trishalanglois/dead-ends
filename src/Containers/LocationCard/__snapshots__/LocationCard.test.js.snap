// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationCard should match the snapshot 1`] = `
<aside
  className="card"
>
  <div
    className="card-title-container"
  >
    <h3
      className="card-title"
    >
      Ada Cemetery
    </h3>
    <div
      className="button-wrapper"
    >
      <button
        className="map-button"
        onClick={[Function]}
      >
        Add to Favorites
      </button>
      <Link
        to="location/Ada Cemetery"
      >
        <button
          className="card-button"
          onClick={[Function]}
        >
          View Map
        </button>
      </Link>
    </div>
  </div>
  <p
    className="card-description"
  >
    BOO!
  </p>
</aside>
`;
